octane/evmengine/keeper/abci.go

```diff
@@ -300,7 +300,8 @@ func (k *Keeper) startBuild(ctx context.Context, appHash common.Hash, timestamp
 		FinalizedBlockHash: headHash,
 	}
 
-	withdrawals, err := k.EligibleWithdrawals(ctx)
+	callCtx := sdk.UnwrapSDKContext(ctx).WithBlockHeight(height)
+	withdrawals, err := k.EligibleWithdrawals(callCtx) //nolint:contextcheck // no new ctx
```

Comment:
FYI: This also worked for me:
https://github.com/mitosis-org/omni/commit/6764a63df55b485950470a049352b722971f9fa7

Reply:
Thanks @taeguk but using all available withdrawals is incorrect. Due to some octane-specific implementation details, when the proposal (containing the withdrawals) is created, we have no effects of BeginBlockers on the consensus state yet, but when we execute the proposal we do run  the BeginBlockers which might create new withdrawals which we didn't observe during the proposal validation. Hence, we have to skip the current block height entirely.

Reply:
Yeah you're right. I was wrong. Using all available withdrawals is incorrect in the manner of general usage.

Fortunately, I called `InsertWithdrawal()` only from `EndBlocker` and event processors, not from `BeginBlocker`, so this was fine for my limited use case. It worked fine for now, but it could be bug-prone in future code changes.